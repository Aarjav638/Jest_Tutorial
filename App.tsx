import {View, Text} from 'react-native';
import React from 'react';
import CustomButton from './components/CustomButton';

const App = () => {
  const [state, setState] = React.useState('Initial State');
  return (
    <View>
      <Text>Testing App</Text>
      <CustomButton
        onPress={() => {
          console.log('Button Pressed');
          setState('Button Pressed');
        }}
        label="My Custom Button"
      />
      <Text testID="State">{state}</Text>
    </View>
  );
};

export default App;
