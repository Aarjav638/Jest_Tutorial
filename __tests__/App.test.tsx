import {render, fireEvent} from '@testing-library/react-native';
import React from 'react';
import App from '../App';

test('renders the UI correctly', () => {
  const tree = render(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Changing States', () => {
  const {getByTestId, queryByTestId} = render(<App />);

  const state = queryByTestId('State');

  expect(state?.props.children).toBe('Initial State');

  const button = getByTestId('MyCustomButton');

  fireEvent.press(button);

  expect(state?.props.children).toBe('Button Pressed');
});
