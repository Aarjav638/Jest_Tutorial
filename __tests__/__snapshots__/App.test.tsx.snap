// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI correctly 1`] = `
<View>
  <Text>
    Testing App
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="MyCustomButton"
  >
    <Text>
      My Custom Button
    </Text>
  </View>
  <Text
    testID="State"
  >
    Initial State
  </Text>
</View>
`;
